pgsql-flexible-server.tf

```diff
@@ -130,12 +137,12 @@ resource "azurerm_postgresql_flexible_server_active_directory_administrator" "pg
 }
 
 resource "null_resource" "set-user-permissions-additionaldbs" {
-  count = var.enable_read_only_group_access ? length(local.db_reader_user) : 0
+  for_each = var.enable_read_only_group_access ? { for index, db in var.pgsql_databases : db.name => db } : {}
```

Comment:
@cpareek could it be reviewed to confirm that is the simplest way to write this?

Reply:
sure - I have checked with Tim on this too as we wanted to only run the block if the enable_read_only_group_access is true, the for_each is taken from existing code here..
https://github.com/hmcts/terraform-module-postgresql-flexible/blob/b5cfd03383426c5eb4a6784773ab9ea4a94100f8/pgsql-flexible-server-db.tf#L2